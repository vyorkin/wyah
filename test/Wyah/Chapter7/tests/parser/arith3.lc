-- Arithmetic
let succ n f x = f (n f x);
