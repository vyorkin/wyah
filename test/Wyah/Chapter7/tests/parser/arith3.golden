Program [Decl "succ" (ELam (Var "n") (ELam (Var "f") (ELam (Var "x") (EApp (EVar (Var "f")) (EApp (EApp (EVar (Var "n")) (EVar (Var "f"))) (EVar (Var "x")))))))]